src/main/java/com/salesforce/WriteTopic.java

```diff
@@ -83,13 +83,13 @@ public Exception call() {
 
 
             while (keepProducing) {
-                produceMessageTimeSecs.time(() -> {
-                    // TODO: Get this from properties
-                    for (int i = 0; i < numMessagesToSendPerBatch; i++) {
-                        kafkaProducer.send(new ProducerRecord<>(topicName, topicId, i));
-                        log.debug("{}: Produced message {}", formatter.format(new Date()), topicId);
-                    }
-                });
+                // TODO: Get this from properties
+                for (int i = 0; i < numMessagesToSendPerBatch; i++) {
+                    Histogram.Timer requestTimer = produceMessageTimeSecs.startTimer();
```

Comment:
Won't this cause memory to explode because you'll now have a separate timer for every single message?